### PR TITLE
Expand isAxiosError types

### PR DESCRIPTION
**Pull request by Argeento**
`isAxiosError` returns `AxiosError<any>`

This change allows to pass type for `AxiosError`

```ts
// Old behavior #1
if (axios.isAxiosError(error)) {
   error.response?.data.message // any
}
// Old behavior #2
if (axios.isAxiosError(error)) {
   (error as AxiosError<MyErrorInterface>).response?.data.message // string | undefined
}

// Behavior after this change #1 (Non-breaking change)
if (axios.isAxiosError(error)) {
   error.response?.data.message // any
}
// Behavior after this change #2 (Desired behavior)
if (axios.isAxiosError<MyErrorInterface>(error)) {
   error.response?.data.message // string | undefined
}

```

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -213,7 +213,7 @@ export interface AxiosStatic extends AxiosInstance {
   isCancel(value: any): boolean;
   all<T>(values: Array<T | Promise<T>>): Promise<T[]>;
   spread<T, R>(callback: (...args: T[]) => R): (array: T[]) => R;
-  isAxiosError(payload: any): payload is AxiosError;
+  isAxiosError<T = any>(payload: any): payload is AxiosError<T>;
```
Comment (arthurfiorette) [inline index.d.ts:216]: AxiosError has two type parameters, I think including both of them would be better.

https://github.com/axios/axios/blob/ea0d9c69f78070dae841c66efd7263e246cab845/index.d.ts#L139

**Review by arthurfiorette — CHANGES_REQUESTED** [review]

File: index.d.ts (hunk 1 above)
Comment (Argeento) [reply index.d.ts]: Good point - thanks!

**Comment by latipun7** [comment]
This PR is duplicate of long ongoing PR #3816 (I can drop the formatting commit if the maintainer didn't want it)

**Review by kirillku — APPROVED** [review]

**Comment by jasonsaayman** [comment]
Closing as the types have already been fixed

**Comment by merlindru** [comment]
@jasonsaayman isAxiosError still doesn't take generics 😕 

I got here by #3815

**Pull request merged**